client/my-sites/site-settings/jetpack-credentials/credentials-form/index.jsx

```diff
@@ -0,0 +1,239 @@
+/** @format */
+/* eslint-disable */
+/**
+ * External dependendies
+ */
+import React, { Component } from 'react';
+import { get, isEmpty } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormSelect from 'components/forms/form-select';
+import FormTextInput from 'components/forms/form-text-input';
+import FormLabel from 'components/forms/form-label';
+import FormTextArea from 'components/forms/form-textarea';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormPasswordInput from 'components/forms/form-password-input';
+
+export class CredentialsForm extends Component {
+	state = {
+		showPrivateKeyField: false,
+		form: {
+			protocol: this.props.protocol,
+			host: this.props.host,
+			port: this.props.port,
+			user: this.props.user,
+			pass: this.props.pass,
+			abspath: this.props.abspath,
+			kpri: this.props.kpri,
+		},
+		formErrors: {
+			host: false,
+			port: false,
+			user: false,
+			pass: false,
+			abspath: false,
+		},
+	};
+
+	handleFieldChange = event => {
+		this.setState( {
+			form: { ...this.state.form, [ event.target.name ]: event.target.value },
+			formErrors: { ...this.state.formErrors, [ event.target.name ]: false },
+		} );
+	};
+
+	handleSubmit = () => {
+		const { siteId, updateCredentials, translate } = this.props;
+
+		const payload = {
+			...this.state.form,
+			role: 'main',
+		};
+
+		const errors = Object.assign(
+			! payload.host && { host: translate( 'Please enter a valid server address.' ) },
+			! payload.port && { port: translate( 'Please enter a valid server port.' ) },
+			isNaN( payload.port ) && { port: translate( 'Port number must be numeric.' ) },
+			! payload.user && { user: translate( 'Please enter your server username.' ) },
+			! payload.pass && { pass: translate( 'Please enter your server password.' ) },
+			! payload.abspath && { abspath: translate( 'Please enter a valid upload path.' ) }
+		);
+
+		return isEmpty( errors )
+			? updateCredentials( siteId, payload )
+			: this.setState( { formErrors: errors } );
+	};
+
+	togglePrivateKeyField = () =>
+		this.setState( { showPrivateKeyField: ! this.state.showPrivateKeyField } );
+
+	render() {
+		const { formIsSubmitting, onCancel, translate } = this.props;
+
+		const { showPrivateKeyField, formErrors } = this.state;
+
+		return (
+			<FormFieldset className="credentials-form">
+				<table>
+					<tbody>
+						<tr>
+							<td colSpan="2" className="credentials-form__protocol-field">
+								<FormLabel>
+									<div>{ translate( 'Credential Type' ) }</div>
+									<FormSelect
+										name="protocol"
+										value={ get( this.state.form, 'protocol', 'ssh' ) }
+										onChange={ this.handleFieldChange }
+										disabled={ formIsSubmitting }
+									>
+										<option value="ssh">{ translate( 'SSH' ) }</option>
```

Comment:
Let's keep the protocols untranslated. Translated terms will just create confusion.